named-snapshot.tmux

```diff
@@ -8,49 +8,78 @@ source "$CURRENT_DIR/scripts/variables.sh"
 bind_prompt_key() {
   local action="$1"
   local key="$2"
+  local switch_client="$4"
+  local client_name=""
+
   if [[ "$action" == "save" ]]; then
     local prompt='Save Snapshot as:'
   else
     local prompt='Restore Snapshot:'
   fi
-  tmux bind-key "$key" command-prompt -p "$prompt" "run-shell '$CURRENT_DIR/scripts/$action-snapshot.sh %1'"
```

Comment:
suggestion: could you refactor this into a reusable function call?

```sh
bind_tmux_key() {
  local key_table="$1"
  local key="$2"
  local prompt="$3"
  local action="$4"
  if [ -n "$prompt" ]; then
    tmux bind-key -T "$key_table" "$key" command-prompt -p "$prompt" "run shell '$CURRENT_DIR/scripts/$action-snapshot.sh %1'"
  else
    tmux bind-key -T "$key_table" "$key" run-shell "$CURRENT_DIR/scripts/$action-snapshot.sh $name"
  fi
}

bind_prompt_key() {
  # ...
  local key_table="root"
  if [ -n "$switch_client" ]; then
    key_table="$(echo "$switch_client" | cut -d':' -f2)"
  fi
  bind_tmux_key "$key_table" "$key" "$prompt" "$action"
  # ...
}
```

Reply:
Replied here: https://github.com/spywhere/tmux-named-snapshot/pull/5#issuecomment-2207154911